### PR TITLE
6651 and Dnc.HostGenerator

### PR DESCRIPTION
**Pull request by rseanhall**
Fixes https://github.com/wixtoolset/issues/issues/6651

### CODE REVIEW

File: src/ext/Bal/test/examples/WPFCoreMBA/Example.WPFCoreMBA.csproj [hunk 1]
```diff
@@ -3,22 +3,8 @@
...
-
-  <Target Name="GitVersion">
-    <PropertyGroup>
-      <GitBaseVersionMajor>42</GitBaseVersionMajor>
-      <GitBaseVersionMinor>42</GitBaseVersionMinor>
-      <GitBaseVersionPatch>42</GitBaseVersionPatch>
-      <GitCommits>42</GitCommits>
-      <GitSha>abc123def456</GitSha>
-    </PropertyGroup>
-  </Target>
```
Comment (rseanhall) [inline src/ext/Bal/test/examples/WPFCoreMBA/Example.WPFCoreMBA.csproj:23]: @robmen I hope this wasn't required for anything 😄

Comment (robmen) [reply src/ext/Bal/test/examples/WPFCoreMBA/Example.WPFCoreMBA.csproj:23]: Can't imagine what. 😄

**Comment by robmen** [comment]
Curious, what drove you to the source generator? Was there an error or anything about needing to support trimming? Or just preparing for the future?

**Comment by rseanhall** [comment]
With the source generator, trimming the BA should Just Work. With the old way using reflection, it would be very cumbersome and error-prone to enable trimming and not break the BA. The BA would build fine but fail at runtime.

**Pull request merged**